R/versioning.R

```diff
@@ -0,0 +1,97 @@
+#' Turn bucket versioning on or off, check status (default), or
+#' list both live and archived versions of objects in the bucket
+#' @param action "status", "enable", "disable", or "list"
+#' @param bucket gcs bucket
+#'
+#' @return versioned_objects list #only if "list" action
+#' @importFrom googleAuthR gar_api_generator
+#' @family managementAPI functions
+#' @export
+gcs_version_bucket <- function(bucket, action = "status") {
+  
+  # support the case where user enters bucket with gs:// prefix
+  bucket <- gsub("^gs://", "", bucket)
```

Comment:
```suggestion
  bucket <- as.bucket_name(bucket)
```

The other bucket functions also support you passing in a bucket object (returned from `gcs_get_bucket()` say) as well as a string of the name.  You can have this behaviour too by using `  bucket <- as.bucket_name(bucket)` at the start.  I think the `gs://` is also supported, but if it is not, the check can be added to the `as.bucket_name()` function so all functions have the same behaviour.

Reply:
Noted. FYI - "gs://" prefix is not current supported for function ```gcs_get_bucket(bucket)```

Reply:
Oh I see ```as.bucket_name()``` is part of all bucket functions, so adjusting that one function should take of things. I can have a look at that.